### PR TITLE
Refresh list of remote kernels if a notebook is already open

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #7385

### CODE REVIEW

**Comment by joyceerhl** [comment]
VS Code used to fire onDidOpenNotebookDocument for already-open notebook documents. This behavior changed in this last iteration and is probably the reason for this problem, I also fixed a problem with export not working that Rong found which was caused by this. I think we should check with VS Code re: whether this was an intentional change

**Review by rchiodo — APPROVED** [review]

File: src/client/datascience/notebook/notebookControllerManager.ts [hunk 1]
```diff
@@ -112,9 +112,7 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         // Sign up for document either opening or closing
         this.notebook.onDidOpenNotebookDocument(this.onDidOpenNotebookDocument, this, this.disposables);
         // If the extension activates after installing Jupyter extension, then ensure we load controllers right now.
-        if (this.isLocalLaunch) {
-            this.notebook.notebookDocuments.forEach((notebook) => this.onDidOpenNotebookDocument(notebook).catch(noop));
-        }
+        this.notebook.notebookDocuments.forEach((notebook) => this.onDidOpenNotebookDocument(notebook).catch(noop));
```
Comment (IanMatthewHuff) [inline src/client/datascience/notebook/notebookControllerManager.ts:115]: Change makes sense, do we know why it was gated on isLocalLaunch in the first place? That doesn't seem relevant to me. Just making sure we're not regressing on some reason this was only local before.

**Review by IanMatthewHuff — APPROVED** [review]

**Review by DavidKutu — APPROVED** [review]

File: src/client/datascience/notebook/notebookControllerManager.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/client/datascience/notebook/notebookControllerManager.ts:115]: Back then I wasn't sure how remote worked, and didn't want to break remote.
I think I fixed this at the last minute before a release, hence didn't want to make unnecessary changes
